src/hdmf/build/map.py

```diff
@@ -1337,6 +1343,20 @@ def __get_cls_dict(self, base, addl_fields):
                     new_args.append(f)
                 if issubclass(dtype, (Container, Data, DataRegion)):
                     fields.append({'name': f, 'child': True})
+                    if hasattr(field_spec, 'quantity') and field_spec.quantity == '*':
```

Comment:
Check for quantity '+' (i.e. _one_ or more) as well

Reply:
```suggestion
                    if hasattr(field_spec, 'quantity') and field_spec.quantity in ('*', '+'):
```